pkg/unidling/metrics/metrics.go

```diff
@@ -0,0 +1,76 @@
+package prometheus
+
+import (
+	"sync"
+
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+var (
+	unidlingCountDesc = prometheus.NewDesc(
+		"openshift_unidling_count",
+		"The number of times an unidling occurred",
+		nil,
+		nil,
+	)
+	uc = unidlingCollector{}
+)
```

Comment:
The Prometheus SDK has a built-in Counter that is much simpler to use and handles atomic increments. Kubernetes component-base has a wrapper around this which makes it compatible with MustRegister. See https://github.com/kubernetes/component-base/blob/v0.22.3/metrics/counter.go#L41.

Metrics should adhere to Prometheus standards for naming. Counts that do not have units should use the `total` suffix. See https://prometheus.io/docs/practices/naming/.

```suggestion
var (
	unidlingCountMetric = metrics.NewCounter(&CounterOps{
	             Name: "openshift_unidle_events_total",
	             Help: "Total count of unidling events observed by the unidling controller"
	         },
	)
)

func IncrementUnidlingCount() {
    unidlingCountMetric.Inc()
}
```

Reply:
@adambkaplan Thank you for the review. Appreciated. I originally did not think I could use Prometheus counter with the existing way this component publishes metrics. Thank you so much for dramatically simplifying my PR.